R/FilteredData.R

```diff
@@ -764,48 +764,11 @@ FilteredData <- R6::R6Class( # nolint
             }
           )
 
-          # rather than regenerating the UI dynamically for the dataset filtering,
-          # we instead choose to hide/show the elements
-          # the filters for this dataset are just hidden from the UI, but still applied
-          # optimization: we set `priority = 1` to execute it before the other
-          # observers (default priority 0), so that they are not computed if they are hidden anyways
           observeEvent(active_datanames(),
-            priority = 1,
             {
-              logger::log_trace(
-                "FilteredData$srv_filter_panel@1 active datanames: { paste(active_datanames(), collapse = \" \") }"
-              )
-              if (length(active_datanames()) == 0 || is.null(active_datanames())) {
-                # hide whole module UI when no datasets or when NULL
-                shinyjs::hide("filter_panel_whole")
-                shinyjs::runjs('$("#teal_secondary_col").hide();
-                             $("#teal_primary_col").attr("class", "col-sm-12").resize();')
-              } else {
-                shinyjs::show("filter_panel_whole")
-                shinyjs::runjs('if (filter_open) {
-              $("#teal_primary_col").attr("class", "col-sm-9").resize();
-              $("#teal_secondary_col").show();}')
-
-                # selectively hide / show to only show `active_datanames` out of all datanames
-                lapply(
-                  self$datanames(),
-                  function(dataname) {
-                    id_add_filter <- private$get_ui_add_filter_id(dataname)
-                    id_filter_dataname <- private$get_ui_id(dataname)
-
-                    if (dataname %in% active_datanames()) {
-                      # shinyjs takes care of the namespace around the id
-                      shinyjs::show(id_add_filter)
-                      shinyjs::show(id_filter_dataname)
-                    } else {
-                      shinyjs::hide(id_add_filter)
-                      shinyjs::hide(id_filter_dataname)
-                    }
-                  }
-                )
-              }
+              private$hide_inactive_datasets(active_datanames)
             },
-            ignoreNULL = FALSE
+            priority = 1
```

Comment:
why is it needed? I think it's fine without `priority`

Reply:
it will work without the priority = 1, I assumed Konrad made some research that it boost performance